glotaran/analysis/problem.py

```diff
@@ -0,0 +1,840 @@
+import collections
+import itertools
+import typing
+
+import numpy as np
+import xarray as xr
+
+from glotaran.analysis.nnls import residual_nnls
+from glotaran.analysis.variable_projection import residual_variable_projection
+from glotaran.parameter import ParameterGroup
+
+from .scheme import Scheme
+
+T_DatasetDescriptor = typing.TypeVar("glotaran.model.DatasetDescriptor")
+T_Model = typing.TypeVar("glotaran.model.Model")
```

Comment:
This will cause 'not valid as a type' see #456 for a reference how it should be done.